platform-api/src/main/java/io/datacater/core/deployment/K8Deployment.java

```diff
@@ -0,0 +1,247 @@
+package io.datacater.core.deployment;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.datacater.core.pipeline.PipelineEntity;
+import io.datacater.core.stream.StreamEntity;
+import io.fabric8.kubernetes.api.model.*;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.fabric8.kubernetes.api.model.apps.DeploymentStatus;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Singleton;
+import org.jboss.logging.Logger;
+
+@Singleton
+public class K8Deployment {
+  private static final Logger LOGGER = Logger.getLogger(K8Deployment.class);
+  private final KubernetesClient client;
+  private final K8ConfigMap k8ConfigMap;
+  private final K8NameSpace k8NameSpace;
+
+  public K8Deployment(KubernetesClient client) {
+    this.client = client;
+    this.k8NameSpace = new K8NameSpace(client);
+    this.k8ConfigMap = new K8ConfigMap(client);
+  }
+
+  public String create(
+      DatacaterDeployment dcDeployment,
+      PipelineEntity pe,
+      StreamEntity streamIn,
+      StreamEntity streamOut) {
+    final String name = getName(dcDeployment.name());
+    final String volumeName = name + StaticConfig.VOLUME_NAME_SUFFIX;
+    k8NameSpace.create();
+
+    if (!pe.equals(new PipelineEntity())) {
+      k8ConfigMap.getOrCreate(name, pe);
+    }
+
+    if (exists(name)) {
+      return name;
+    }
+
+    List<EnvVar> variables = getEnvironmentVariables(dcDeployment, streamIn, streamOut);
+
+    var deployment =
+        new DeploymentBuilder()
+            .withNewMetadata()
+            .withName(name)
```

Comment:
I am not sure whether we should let users name Kubernetes resources. Let me think about it for a bit :)